src/data_validation/data_schema.py

```diff
@@ -0,0 +1,142 @@
+def convert_dataFrame(pdFrame: pd.DataFrame) -> dict:
+    """Convert a pandas dataframe into a dictionary oriented by
+    index. This makes the keys in the dictionary the row index
+    of the dataframe, and the values are a dictionary containing
+    the other key-value information.
+
+    Arguments:
+        pdFrame -- the pandas dataframe to be converted
+
+    Returns:
+        A dict: dict object oriented by index
+    """
+    pd_dict = pdFrame.to_dict(orient="index")
+    return pd_dict
+
+
+def is_nan(value) -> bool:
+    """Takes in a value and returns a boolean indicating
+    whether it is a 'not a number' or not.
+
+    Arguments:
+        value -- Any value
+
+    Returns:
+        A bool: boolean indicating whether the value is
+        'not a number' or not, as determined by the 'math'
+        module.
+    """
+    return math.isnan(float(value))
+
+
+def reformat_tomlDict(pdDict: dict) -> dict:
+    """Creates a dictionary suitable to be converted
+    into a toml file. Takes an index oriented
+    dictionary as input and creates a new dictionary
+    at as
+
+    Arguments:
+        pdDict -- a dictionary
+
+    Returns:
+        A dict: dictionary ready to be used to create
+        a toml file.
+    """
+    newDict = {}
+    tomlDict = {}
+
+    # Loop over input dictionary to create a sub dictionary
+    for key in pdDict:
+        newDict[str(key)] = pdDict[key]
+
+        subDict1 = newDict[str(key)]
+        var = subDict1.pop("Field Name (as it appears in dataset)")
+
+        tomlDict[var] = subDict1
+
+    # Loop over each key in sub-dictionary and reformat values for usability
+    for key in tomlDict:
+
+        subDict2 = tomlDict[key]
+
+        subDict2["description"] = subDict2.pop("Description")
+        subDict2["data_type"] = subDict2.pop(
+            """Data Type (Numeric integer/Numeric float (or decimal)
+            /Text/Categorical/Boolean (True or False, 1 or 0))"""
+        )
+        subDict2["nullable"] = subDict2.pop(
+            "Nullable (is it acceptable to have a null value? Acceptable = Yes)"
+        )
+
+        acceptable_values_str = str(subDict2["Acceptable Values (>0 or 0 – 1,000,000)"])
+        acceptable_values_list = acceptable_values_str.split()
+
+        subDict2["min_acceptable_value"] = acceptable_values_list[0]
+        subDict2["max_acceptable_value"] = acceptable_values_list[-1].replace(",", "")
+
+        if is_nan(subDict2["min_acceptable_value"]):
+            subDict2["min_acceptable_value"] = acceptable_values_list[0]
+        elif is_nan(subDict2["max_acceptable_value"]):
+            subDict2["max_acceptable_value"] = acceptable_values_list[-1]
+        else:
+            subDict2["min_acceptable_value"] = int(acceptable_values_list[0])
+            subDict2["max_acceptable_value"] = int(
+                acceptable_values_list[-1].replace(",", "")
+            )
+
+        subDict2.pop("Acceptable Values (>0 or 0 – 1,000,000)")
+
+        tomlDict[key] = subDict2
```

Comment:
File since deleted.

Appreciate the advice and have been keeping it in mind since. Trying to use dataframes and Pandas-native functions where possible (or other native functions for that matter). Seeking advice where I am unable to resolve an issue using native functions.
Where it is sensible to do so I have tried to comment my logic for future maintainers and for PRs (and myself).